src/UseCases/AddDonation/AddDonationPolicyValidator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\UseCases\AddDonation;
+
+use WMDE\Fundraising\Frontend\Validation\AmountPolicyValidator;
+use WMDE\Fundraising\Frontend\Validation\ConstraintViolation;
+use WMDE\Fundraising\Frontend\Validation\TextPolicyValidator;
+use WMDE\Fundraising\Frontend\UseCases\AddDonation\AddDonationValidationResult as Result;
+
+/**
+ * @license GNU GPL v2+
+ * @author Gabriel Birke < gabriel.birke@wikimedia.de >
+ */
+class AddDonationPolicyValidator {
+
+	private $amountPolicyValidator;
+	private $textPolicyValidator;
+
+	public function __construct( AmountPolicyValidator $amountPolicyValidator, TextPolicyValidator $textPolicyValidator ) {
+		$this->amountPolicyValidator = $amountPolicyValidator;
+		$this->textPolicyValidator = $textPolicyValidator;
+	}
+
+	public function needsModeration( AddDonationRequest $request ): bool {
+		$violations = array_merge(
+			$this->getAmountViolations( $request ),
+			$this->getBadWordViolations( $request )
+		);
+
+		return !empty( $violations );
+	}
+
+	private function getBadWordViolations( AddDonationRequest $request ): array {
+		if ( $request->donorIsAnonymous() ) {
+			return [];
+		}
+
+		return array_merge(
+			$this->getPolicyViolationsForField( $request->getDonorFirstName(), Result::SOURCE_DONOR_FIRST_NAME ),
+			$this->getPolicyViolationsForField( $request->getDonorLastName(), Result::SOURCE_DONOR_LAST_NAME ),
+			$this->getPolicyViolationsForField( $request->getDonorCompany(), Result::SOURCE_DONOR_COMPANY ),
+			$this->getPolicyViolationsForField( $request->getDonorStreetAddress(), Result::SOURCE_DONOR_STREET_ADDRESS ),
+			$this->getPolicyViolationsForField( $request->getDonorPostalCode(), Result::SOURCE_DONOR_POSTAL_CODE ),
```

Comment:
Just a remark/question, out of scope of this PR: On the client side we restrict input to numeric strings. Do you think, we should do this in the server side validation as well? This would make policy checks on this field obsolete.